policy/modules/system/systemd.fc

```diff
@@ -84,11 +84,11 @@ HOME_DIR/\.local/share/systemd(/.*)?		gen_context(system_u:object_r:systemd_data
 /run/nologin	--	gen_context(system_u:object_r:systemd_sessions_runtime_t,s0)
 
 /run/user/%{USERID}/systemd	-d	gen_context(system_u:object_r:systemd_user_runtime_t,s0)
-/run/user/%{USERID}/systemd/generator(/.*)?		gen_context(system_u:object_r:systemd_user_runtime_unit_t,s0)
-/run/user/%{USERID}/systemd/generator\.early(/.*)?		gen_context(system_u:object_r:systemd_user_runtime_unit_t,s0)
-/run/user/%{USERID}/systemd/generator\.late(/.*)?		gen_context(system_u:object_r:systemd_user_runtime_unit_t,s0)
-/run/user/%{USERID}/systemd/transient(/.*)?		gen_context(system_u:object_r:systemd_user_runtime_unit_t,s0)
-/run/user/%{USERID}/systemd/user(/.*)?		gen_context(system_u:object_r:systemd_user_runtime_unit_t,s0)
+/run/user/%{USERID}/systemd/generator(/.*)?		gen_context(system_u:object_r:systemd_user_transient_unit_t,s0)
+/run/user/%{USERID}/systemd/generator\.early(/.*)?		gen_context(system_u:object_r:systemd_user_transient_unit_t,s0)
+/run/user/%{USERID}/systemd/generator\.late(/.*)?		gen_context(system_u:object_r:systemd_user_transient_unit_t,s0)
+/run/user/%{USERID}/systemd/transient(/.*)?		gen_context(system_u:object_r:systemd_user_transient_unit_t,s0)
+/run/user/%{USERID}/systemd/user(/.*)?		gen_context(system_u:object_r:systemd_user_transient_unit_t,s0)
```

Comment:
I've been debating this to myself for a while. Unlike systemd --system, I'm not sure that systemd --user gains much benefit from differentiating between transient and runtime units. One thing for certain is in the above lines, all but the transient dir is a runtime unit.

Reply:
I agree I don't think there's much difference between the two as far as the policy is concerned, but I think that the fact that there is a difference is important enough to treat them differently in case anything changes in the future.

What I really want to do is redefine the user "runtime" unit that is already in the policy so that we have the appropriate types for both the runtime and transient units.

Reply:
I had another look at the systemd docs about runtime vs transient units and I think we should do this instead: keep the current naming as-is, but add a new `systemd_user_transient_unit_t` type for `/run/user/%{USERID}/systemd/transient` and adjust the rules accordingly.

What do you think?

Reply:
Sounds good to me.